mqtt/mqtt-broker/src/broker.rs

```diff
@@ -807,23 +808,24 @@ where
             );
 
             let client_id = connreq.client_id().clone();
-            let (state, client_info_, _will, handle) = current_connected.into_parts();
-            let old_session =
-                Session::new_disconnecting(client_id.clone(), client_info_, None, handle);
-            let (new_session, session_present) =
-                if let proto::ClientId::IdWithExistingSession(_) = connreq.connect().client_id {
-                    debug!(
-                        "moving persistent session to this connection for {}",
-                        client_id
-                    );
-                    let new_session = Session::new_persistent(auth_id, connreq, state);
-                    (new_session, true)
-                } else {
-                    info!("cleaning session for {}", client_id);
-                    let state = SessionState::new(client_id.clone(), self.config.session().clone());
-                    let new_session = Session::new_transient(auth_id, connreq, state);
-                    (new_session, false)
-                };
+            let (state, _will, handle) = current_connected.into_parts();
+            let old_session = Session::new_disconnecting(state.client_info().clone(), None, handle);
+            let (new_session, session_present) = if let proto::ClientId::IdWithExistingSession(_) =
+                connreq.connect().client_id
+            {
+                debug!(
+                    "moving persistent session to this connection for {}",
+                    client_id
+                );
+                let new_session = Session::new_persistent(auth_id, connreq, state);
```

Comment:
since we have auth_id in the state, should we simplify this a bit?
```suggestion
                let new_session = Session::new_persistent(connreq, state);
```